src/Anonymization/Anonymizer/AnonymizerRegistry.php

```diff
@@ -45,42 +51,66 @@ public function addPath(array $paths): void
     /**
      * Get all registered anonymizers classe names.
      *
-     * @return array<string,string>
-     *   Keys are names, values are class names.
+     * @return array<string,AsAnonymizer>
      */
-    public function getAnonymizers(): array
+    public function getAllAnonymizerMetadata(): array
     {
         $this->initialize();
 
-        return $this->anonymizers;
+        return $this->metadata;
     }
 
     /**
-     * Get anonymizer class name.
-     *
-     * @param string $name
-     *   Anonymizer name.
-     *
-     * @return string
-     *   Anonymizer class name.
+     * Create anonymizer instance.
      */
-    public function get(string $name): string
+    public function createAnonymizer(
+        string $name,
+        AnonymizerConfig $config,
+        Options $options,
+        DatabaseSession $databaseSession,
+    ): AbstractAnonymizer {
+        $className = $this->classes[$name] ?? throw new \InvalidArgumentException(\sprintf("Can't find Anonymizer with name : %s, check your configuration.", $name));
+
+        return new $className($config->table, $config->targetName, $databaseSession, $config);
+    }
+
+    /**
+     * Get anonymizer metadata.
+     */
+    public function getAnonymizerMetadata(string $name): AsAnonymizer
     {
-        $this->initialize();
+        return $this->metadata[$name] ?? throw new \InvalidArgumentException(\sprintf("Can't find Anonymizer with name : %s, check your configuration.", $name));
```

Comment:
```suggestion
        return $this->getAllAnonymizerMetadata()[$name] ?? throw new \InvalidArgumentException(\sprintf("Can't find Anonymizer with name : %s, check your configuration.", $name));
```

Reply:
I forgot the `$this->initialize();` call...

Reply:
Fixed!